onnx/defs/schema.cc

```diff
@@ -822,18 +829,67 @@ const FunctionProto* OpSchema::GetFunction() const {
   return GetFunctionWithOpsetVersion(since_version_);
 }
 
-const FunctionProto* OpSchema::GetFunctionWithOpsetVersion(int opset_version) const {
-  return const_cast<OpSchema*>(this)->GetFunctionWithOpsetInternal(opset_version);
+const FunctionProto*
+OpSchema::GetFunctionWithOpsetVersion(int opset_version, const std::string& domain, bool fail_on_invalid_op) const {
+  return const_cast<OpSchema*>(this)->GetFunctionWithOpsetInternal(opset_version, domain, fail_on_invalid_op);
 }
 
-FunctionProto* OpSchema::GetFunctionWithOpsetInternal(int opset_version) {
-  std::map<int, std::shared_ptr<FunctionProto>>::iterator it =
-      opset_version_to_function_body_.lower_bound(opset_version);
-  if (it != opset_version_to_function_body_.cend())
-    return it->second.get();
+void OpSchema::ValidateReferencedOpsInFunciton(
+    const FunctionProto* function,
+    int requested_opset_version,
+    int function_since_version,
+    const std::string& domain,
```

Comment:
I suggest dropping this `domain` parameter. (Or, the longer term solution is to replace the `int opset_version` parameter by a map from domain to version.)